webservices/common/views.py

```diff
@@ -66,42 +66,33 @@ class ItemizedResource(ApiResource):
     year_column = None
     index_column = None
     filters_with_max_count = []
+    union_all_fields = []
     max_count = 10
+    secondary_index_options = []
 
     def get(self, **kwargs):
-        """Get itemized resources. If multiple values are passed for `committee_id`,
-        create a subquery for each and combine with `UNION ALL`. This is necessary
+        """Get itemized resources.
+
+        If multiple values are passed for `committee_id` or `two_year_transaction_period`,
+        create a subquery for each value and combine with `UNION ALL`. This is necessary
         to avoid slow queries when one or more relevant committees has many
         records.
         """
-        if kwargs.get("last_index"):
-            if all(
-                kwargs.get("last_{}".format(option)) is None
-                for option in self.sort_options
-            ) and not kwargs.get("sort_null_only"):
-                raise exceptions.ApiError(
-                    "When paginating through results, both values from the \
-                    previous page's `last_indexes` object are needed. For more information, \
-                    see https://api.open.fec.gov/developers/. Please add one of the following \
-                    filters to your query: `sort_null_only`=True, {}".format(
-                        ", ".join("`last_" + option + "`" for option in self.sort_options)
-                    ),
-                    status_code=422,
-                )
-        over_limit_fields = [
-            field
-            for field in self.filters_with_max_count
-            if len(kwargs.get(field, [])) > self.max_count
-        ]
-        if over_limit_fields:
-            raise exceptions.ApiError(
-                "Can only specify up to {0} values for `{1}`".format(
-                    self.max_count, "`, `".join(over_limit_fields)
-                ),
-                status_code=422,
-            )
+        self.validate_kwargs(kwargs)
+        # Generate UNION ALL subqueries if querying multiple committee IDs
         if len(kwargs.get("committee_id", [])) > 1:
-            query, count = self.join_committee_queries(kwargs)
+            query, count = self.join_union_subqueries(kwargs, union_field="committee_id")
+            return utils.fetch_seek_page(query, kwargs, self.index_column, count=count)
+        # If 2-year period isn't specified, add all to kwargs
+        if "two_year_transaction_period" in self.union_all_fields and not kwargs.get(
+            "two_year_transaction_period"
+        ):
+            kwargs["two_year_transaction_period"] = range(
+                1976, utils.get_current_cycle() + 2, 2
+            )
+        # Generate UNION ALL subqueries if searching multiple 2-year periods
+        if len(kwargs.get("two_year_transaction_period", [])) > 1:
```

Comment:
Loop through union all fields - allows us to "turn off" two-year union more easily